modules/dmrpp_module/build_dmrpp.cc

```diff
@@ -244,6 +244,45 @@ int main(int argc, char *argv[]) {
             // is fragile. - ndp 6/6/18
             string h5_file_path = BESUtil::assemblePath(bes_data_root, h5_file_name);
 
+            //Use this ifstream file to run a check on the provided file's signature
+            // to see if it is an HDF5 file
+            ifstream file(h5_file_path, ios::binary);
+            if (!file) {
+                cerr << "Error opening file: " << h5_file_path << endl;
+                cerr << "Cause of error: " << strerror(errno) << endl;
```

Comment:
In my first comment, I mentioned error. 
But then I did some reading and discovered that `std:ifstream` does not set errno, like ever.
This prompted me to add my second comment in which I provided some links about `std::ifstream`
 
This:
```
cerr << "Cause of error: " << strerror(errno) << endl;
```
Is not a solution for `std::ifstream`

Reply:
I updated the error checking based on the links you posted and what I found here: [dealing correctly with badbit, failbit, eofbit, and perror()](https://gehrcke.de/2011/06/reading-files-in-c-using-ifstream-dealing-correctly-with-badbit-failbit-eofbit-and-perror/)

Reply:
Well, I got something completely from that article different and I think this code still needs to be rewritten. 

* Don't  use `perror` that just writing to stderr and dow nothing at handle to handle the problem.
* Throw exceptions when things go wrong, otherwise the code just moves on as if everything is fine, and it's not, so more bad things will happen.
*